pkg/kube/watcher.go

```diff
@@ -67,22 +69,31 @@ func (e *EventWatcher) OnUpdate(oldObj, newObj interface{}) {
 	e.onEvent(event)
 }
 
-// Ignore events older than the maxEventAgeSeconds
+// Ignore events older than the maxEventAgeSeconds.
+// Event age is calculated as the time since the event was created.
 func (e *EventWatcher) isEventDiscarded(event *corev1.Event) bool {
-	timestamp := event.LastTimestamp.Time
-	if timestamp.IsZero() {
-		timestamp = event.EventTime.Time
-	}
-	eventAge := time.Since(timestamp)
+	eventCreationTime := event.CreationTimestamp.Time
```

Comment:
When is `event.LastTimestamp` updated? Is it more recent then `event.CreationTimestamp` within a given TTL window?

Reply:
`event.LastTimestamp` is updated when a new event is seen that's similar to the previous events. it can either be same (event creation after last one expired) or more recent than `event.CreationTimestamp` in a given TTL window (if the previous event was within TTL and it was patched).

Reply:
So, does that mean the only cases where creationTime would be more recent is when a new event arrives after the last occurrence of that event in the cache expired? If that is the case, not entirely clear how much difference this PR will make to the dropped events.

Reply:
Technically, yes. But in practice, we see various cases where creationTimestamp is a few mins ahead of lastTimestamp. So, now I am checking for the greater of the two so we pick the most recent value.